### ISSUE TITLE
AttributeError: module 'pkgutil' has no attribute 'ImpImporter'

### ISSUE THREAD

**Issue by kaushalyap** [paul-gauthier/aider#266]
```
❯ python -m pip install aider-chat
Defaulting to user installation because normal site-packages is not writeable
Collecting aider-chat
  Obtaining dependency information for aider-chat from https://files.pythonhosted.org/packages/20/c3/79a22d1b89ebb3429ea6a533574b021fd236afde2c22a8edc199d77343c4/aider_chat-0.14.0-py3-none-any.whl.metadata
  Using cached aider_chat-0.14.0-py3-none-any.whl.metadata (11 kB)
Collecting aiohttp==3.8.4 (from aider-chat)
  Using cached aiohttp-3.8.4.tar.gz (7.3 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Collecting aiosignal==1.3.1 (from aider-chat)
  Using cached aiosignal-1.3.1-py3-none-any.whl (7.6 kB)
Collecting async-timeout==4.0.2 (from aider-chat)
  Using cached async_timeout-4.0.2-py3-none-any.whl (5.8 kB)
Collecting attrs==23.1.0 (from aider-chat)
  Using cached attrs-23.1.0-py3-none-any.whl (61 kB)
Collecting certifi==2023.5.7 (from aider-chat)
  Using cached certifi-2023.5.7-py3-none-any.whl (156 kB)
Collecting charset-normalizer==3.1.0 (from aider-chat)
  Using cached charset_normalizer-3.1.0-py3-none-any.whl (46 kB)
Collecting frozenlist==1.3.3 (from aider-chat)
  Using cached frozenlist-1.3.3.tar.gz (66 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting gitdb==4.0.10 (from aider-chat)
  Using cached gitdb-4.0.10-py3-none-any.whl (62 kB)
Collecting GitPython==3.1.31 (from aider-chat)
  Using cached GitPython-3.1.31-py3-none-any.whl (184 kB)
Requirement already satisfied: idna==3.4 in /usr/lib/python3.12/site-packages (from aider-chat) (3.4)
Collecting markdown-it-py==2.2.0 (from aider-chat)
  Using cached markdown_it_py-2.2.0-py3-none-any.whl (84 kB)
Requirement already satisfied: mdurl==0.1.2 in /usr/lib/python3.12/site-packages (from aider-chat) (0.1.2)
Requirement already satisfied: multidict==6.0.4 in /usr/lib64/python3.12/site-packages (from aider-chat) (6.0.4)
Collecting openai==0.27.6 (from aider-chat)
  Using cached openai-0.27.6-py3-none-any.whl (71 kB)
Collecting prompt-toolkit==3.0.38 (from aider-chat)
  Using cached prompt_toolkit-3.0.38-py3-none-any.whl (385 kB)
Requirement already satisfied: Pygments==2.15.1 in /usr/lib/python3.12/site-packages (from aider-chat) (2.15.1)
Collecting requests==2.30.0 (from aider-chat)
  Using cached requests-2.30.0-py3-none-any.whl (62 kB)
Collecting rich==13.3.5 (from aider-chat)
  Using cached rich-13.3.5-py3-none-any.whl (238 kB)
Collecting smmap==5.0.0 (from aider-chat)
  Using cached smmap-5.0.0-py3-none-any.whl (24 kB)
Collecting tqdm==4.65.0 (from aider-chat)
  Using cached tqdm-4.65.0-py3-none-any.whl (77 kB)
Collecting urllib3==2.0.2 (from aider-chat)
  Using cached urllib3-2.0.2-py3-none-any.whl (123 kB)
Collecting wcwidth==0.2.6 (from aider-chat)
  Using cached wcwidth-0.2.6-py2.py3-none-any.whl (29 kB)
Collecting yarl==1.9.2 (from aider-chat)
  Using cached yarl-1.9.2.tar.gz (184 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting pytest==7.3.1 (from aider-chat)
  Using cached pytest-7.3.1-py3-none-any.whl (320 kB)
Collecting tiktoken==0.4.0 (from aider-chat)
  Using cached tiktoken-0.4.0.tar.gz (25 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting configargparse (from aider-chat)
  Obtaining dependency information for configargparse from https://files.pythonhosted.org/packages/6f/b3/b4ac838711fd74a2b4e6f746703cf9dd2cf5462d17dac07e349234e21b97/ConfigArgParse-1.7-py3-none-any.whl.metadata
  Using cached ConfigArgParse-1.7-py3-none-any.whl.metadata (23 kB)
Requirement already satisfied: PyYAML in /usr/lib64/python3.12/site-packages (from aider-chat) (6.0.1)
Collecting backoff==2.2.1 (from aider-chat)
  Using cached backoff-2.2.1-py3-none-any.whl (15 kB)
Collecting networkx==3.1 (from aider-chat)
  Using cached networkx-3.1-py3-none-any.whl (2.1 MB)
Collecting diskcache==5.6.1 (from aider-chat)
  Using cached diskcache-5.6.1-py3-none-any.whl (45 kB)
Collecting numpy==1.24.3 (from aider-chat)
  Using cached numpy-1.24.3.tar.gz (10.9 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [33 lines of output]
      Traceback (most recent call last):
        File "/usr/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/usr/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/usr/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 112, in get_requires_for_build_wheel
          backend = _build_backend()
                    ^^^^^^^^^^^^^^^^
        File "/usr/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 77, in _build_backend
          obj = import_module(mod_path)
                ^^^^^^^^^^^^^^^^^^^^^^^
        File "/usr/lib64/python3.12/importlib/__init__.py", line 90, in import_module
          return _bootstrap._gcd_import(name[level:], package, level)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "<frozen importlib._bootstrap>", line 1381, in _gcd_import
        File "<frozen importlib._bootstrap>", line 1354, in _find_and_load
        File "<frozen importlib._bootstrap>", line 1304, in _find_and_load_unlocked
        File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
        File "<frozen importlib._bootstrap>", line 1381, in _gcd_import
        File "<frozen importlib._bootstrap>", line 1354, in _find_and_load
        File "<frozen importlib._bootstrap>", line 1325, in _find_and_load_unlocked
        File "<frozen importlib._bootstrap>", line 929, in _load_unlocked
        File "<frozen importlib._bootstrap_external>", line 994, in exec_module
        File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
        File "/tmp/pip-build-env-xe2fwt1g/overlay/lib/python3.12/site-packages/setuptools/__init__.py", line 16, in <module>
          import setuptools.version
        File "/tmp/pip-build-env-xe2fwt1g/overlay/lib/python3.12/site-packages/setuptools/version.py", line 1, in <module>
          import pkg_resources
        File "/tmp/pip-build-env-xe2fwt1g/overlay/lib/python3.12/site-packages/pkg_resources/__init__.py", line 2172, in <module>
          register_finder(pkgutil.ImpImporter, find_on_path)
                          ^^^^^^^^^^^^^^^^^^^
      AttributeError: module 'pkgutil' has no attribute 'ImpImporter'. Did you mean: 'zipimporter'?
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

**Comment by paul-gauthier** [comment]
Thanks for trying aider!

Have you tried using python3.11?

**Comment by kaushalyap** [comment]
> Thanks for trying aider!
> 
> Have you tried using python3.11?

```
❯ python3 --version               
Python 3.12.0rc3
```

**Comment by paul-gauthier** [comment]
As far as I know python 3.12 is not actually released yet? Aider is currently tested on 3.9-3.11, so please try using one of those versions.

**Comment by kaushalyap** [comment]
> As far as I know python 3.12 is not actually released yet? Aider is currently tested on 3.9-3.11, so please try using one of those versions.

I was actually on Fedora 39 Beta thus probably the 3.12rc, I'll try 3.11 and let you know.

**Comment by paul-gauthier** [comment]
I'm going to close this issue for now, but feel free to re-open or file a new issue any time.
